src/Compilers/VisualBasic/Portable/Scanner/ScannerBuffer.vb

```diff
@@ -134,8 +134,78 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return ch
         End Function
 
+        Friend Function TryPeek(ByRef c As Char) As Boolean
+            If _lineBufferOffset >= _bufferLen Then Return False
+            Dim page = _curPage
+            Dim pos = _lineBufferOffset
+            Dim ch = page._arr(pos And PAGE_MASK)
+            Dim start = page._pageStart
+            Dim expectedStart = pos And NOT_PAGE_MASK
+            If start <> expectedStart Then
+                page = GetPage(pos)
+                ch = page._arr(pos And PAGE_MASK)
+            End If
+            c = ch
+            Return True
+        End Function
+
+        Friend Function TryGet(ByRef c As String) As Boolean
+            If _lineBufferOffset >= _bufferLen Then Return False
+            Dim page = _curPage
+            Dim pos = _lineBufferOffset
+            Dim ch = page._arr(pos And PAGE_MASK)
+            Dim start = page._pageStart
+            Dim expectedStart = pos And NOT_PAGE_MASK
+
+            If start <> expectedStart Then
+                page = GetPage(pos)
+                ch = page._arr(pos And PAGE_MASK)
+            End If
+            c = Intern(ch)
+            Return True
+        End Function
+
+        ' PERF CRITICAL
+        Private Function TryPeek(skip As Integer, ByRef c As Char) As Boolean
+            ' CanGetCharAtOffset( )
+            If _lineBufferOffset + skip >= _bufferLen Then Return False
+            If skip < -MaxCharsLookBehind Then Return False
+            Dim pos = _lineBufferOffset
+            Dim page = _curPage
+            pos += skip
+            Dim ch = page._arr(pos And PAGE_MASK)
+            Dim start = page._pageStart
+            Dim expectedStart = pos And NOT_PAGE_MASK
+            If start <> expectedStart Then
+                page = GetPage(pos)
+                ch = page._arr(pos And PAGE_MASK)
+            End If
+            c = ch
+            Return True
+        End Function
+
+        Private Function IsPeek(at As Integer, isChar As Char) As Boolean
+            Dim c As Char
+            Return TryPeek(at, c) AndAlso (c = isChar)
+        End Function
+
+        Private Function ArePeek(chars As String) As Boolean
+            Return ArePeek(_lineBufferOffset, chars.Length, 1, chars)
```

Comment:
I think this is still wrong. Since "skip" is 1, then "size" needs to be ```chars.Length + 1```
Also, I think the way you had it before was more useful. Skip isn't always 1.